sudokuGUI.py

```diff
@@ -1,532 +1,533 @@
+        self.screen_text = "Sudoku Game"
+        self.start = None
+        self.time = None
+        self.finished = False
+        self.cell_changed = False
+        self.error_msg = False
+        self.menu_buttons = []
+        self.playing_buttons = []
+        self.end_game_buttons = []
+        self.lock_cells = []
+        self.errors = 0
+        self.hints = 3
+        self.font = pygame.font.SysFont('comicsans', 40)
+        self.db = DataBase()
+        self.load_buttons()
+
+
+    def run(self):
+        while self.running:
+            if self.state == 'menu':
+                self.menu_events()
+                self.menu_update()
+                self.menu_draw()
+
+            if self.state == 'playing':
+                self.playing_events()
+                self.playing_update()
+                self.playing_draw()
+
+            if self.state == 'win' or self.state == 'lose':
+                self.game_over_events()
+                self.game_over_update()
+                self.game_over_draw()
+
+        pygame.quit()
+
+
+# Menu state functions #
+
+    # Function that responsible on the menu differen events
+    # Input: Nothing
+    # Output: Nothing
+    def menu_events(self):
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                self.running = False
+                
+            # When the user clicks
+            if event.type == pygame.MOUSEBUTTONDOWN:
+                for button in self.menu_buttons:
+                    if button.highlighted:
+                        button.click()    
+
+
+    # Function that updates the game (clicking on buttons etc)
+    # Input: nothing
+    # Output: Nothing
+    def menu_update(self):
+        self.mouse_pos = pygame.mouse.get_pos()
+        for button in self.menu_buttons:
+            button.update(self.mouse_pos)
+
+
+    # Function that draws the menu with the levels option
+    # Input: Nothing
+    # Output: Nothing
+    def menu_draw(self):
+        self.window.fill(BLACK)
+
+        for button in self.menu_buttons:
+            button.draw(self.window)
+  
+        menu_font = pygame.font.SysFont('comicsans', 100)
+        font = menu_font.render(self.screen_text, False, PINK)
+        self.window.blit(font, [65, 50])
+
+        pygame.display.update()
+        pygame.display.set_caption(self.caption)
+
+
+    # Function that randomly gets a board from the DataBase class and initiate variables by it
+    # Input: The level the user chose
+    # Output: Nothing
+    def get_board(self, level : int):
+        self.grid = self.db.get_rand_board(level)
+        self.finished_board = deepcopy(self.grid)
+        solving(self.finished_board)
+        self.load_board()
+        self.state = 'playing'
+        self.start = time.time()
+
+# Playing state functions # 
+
+    # Function that responsible on the playing differen events
+    # Input: Nothing
+    # Output: Nothing
+    def playing_events(self):
```

Comment:
Functions should only have one objective.
Please split this function into smaller functions.
( This topic is long so call me sometime and ask me about it )